File: Algorithms/Sorting/bubble_sort.py
def bubble_sort(input_list: list):
    # Time complexity: O(n^2)
    # Space complexity: O(1)
    if not isinstance(input_list, list):
        raise TypeError("Wrong type of input")
    for traverse_index in range(len(input_list), 0, -1):
        for i in range(traverse_index-1):
            if input_list[i] > input_list[i+1]:
                input_list[i], input_list[i+1] = \
                    input_list[i+1], input_list[i]
    return input_list


if __name__ == "__main__":
    print("Sorting [1, 5, 3, 2, 9, 8, 6]:",
    bubble_sort([1, 5, 3, 2, 9, 8, 6]))
    print("Sorting [7, 2, 1, 9, 3, 2]:",
    bubble_sort([7, 2, 1, 9, 3, 2]))

File: Algorithms/Sorting/bubble_sort.js
function bubbleSort(array) {
    // Time complexity: O(n^2)
    // Space complexity: O(1)
    for (let trav_index=array.length; trav_index > 0; trav_index--) {
        for (let i=0; i < trav_index-1; i++) {
            if (array[i] > array[i+1]) {
                [array[i], array[i+1]] = [array[i+1], array[i]];
            }
        }
    }
    return array;
}

if (require.main === module) {
    console.log("Sorting [1, 5, 3, 2, 9, 8, 6]:",
    bubbleSort([1, 5, 3, 2, 9, 8, 6]));
    console.log("Sorting [7, 2, 1, 9, 3, 2]:",
    bubbleSort([7, 2, 1, 9, 3, 2]));
}
